### PR TITLE
crew: Retry `make` with single thread if failed

### PR DESCRIPTION
**Pull request by supechicken**
Suggested by @satmandu:

> It would be nice to automatically retry commands which have a -j#{CREW_NPROC} passed to it which fail automatically with -j1 so we can see the command that failed...

Tested on `x86_64`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/supechicken/chromebrew.git CREW_TESTING_BRANCH=retry_make CREW_TESTING=1 crew update
```

### CODE REVIEW

File: lib/package.rb [hunk 1]
```diff
@@ -276,19 +276,39 @@ def self.system(*args, **opt_args)
...
+        # replace placeholder with '-j#' arg and execute the actual command
+        Kernel.system(env, *cmd_args.map { |arg| arg.sub('<<<CREW_NPROC>>>', "-j#{make_threads}") }, **opt_args)
+      else
+        Kernel.system(env, *cmd_args, **opt_args)
+      end
+    rescue RuntimeError => e
+      if modded_make_cmd && make_threads != 1
+        # retry with single thread if command is `make` and is modified by crew
+        warn "Command \"#{cmd_args.map { |arg| arg.sub('<<<CREW_NPROC>>>', "-j#{make_threads}") } .join(' ')}\" failed, retrying with \"-j1\"...".yellow
+        make_threads = 1
```
Comment (uberhacker) [inline lib/package.rb:306]: What does setting this value do if it's not being used later?

Comment (satmandu) [reply lib/package.rb:306]: @supechicken Do you want to answer this so we can get it merged?

Comment (supechicken) [reply lib/package.rb:306]: Sorry for late reply😅 This variable will be reused in line 296-301 after calling `retry` (line 307) in this `rescue` part.

Comment (uberhacker) [reply lib/package.rb:306]: OK, so this acts like a loop then.

Comment (uberhacker) [reply lib/package.rb:306]: What can we do to prevent an endless loop?  Should we limit the number of retries?

Comment (supechicken) [reply lib/package.rb:306]: This retry logic will be skipped when `make_threads` is `1`

Comment (satmandu) [reply lib/package.rb:306]: @uberhacker Any other objections?

**Review by satmandu — APPROVED** [review]

File: lib/package.rb (hunk 1 above)
Comment (uberhacker) [reply lib/package.rb:306]: Bump crew version?

**Pull request merged**